src/main/java/com/example/petbuddybackend/service/chat/session/ChatSessionService.java

```diff
@@ -56,56 +51,42 @@ public void sendMessages(ChatRoom chatRoom, ChatNotification notification) {
                 .forEach((simpSession) -> sendToUser(chatId, caretakerEmail, simpSession.getId(), notification));
     }
 
-    public void onUserJoinChatRoom(String joiningUsername, Long chatId, String subId) {
+    public void onUserJoinChatRoom(String joiningUsername, Long chatId) {
         ChatRoom chatRoom = chatService.getChatRoomById(chatId);
-        chatSessionTracker.addSubscription(subId, chatId);
-        Map<Long, Integer> allChatRoomSessions = countChatRoomSessions(joiningUsername);
+        Map<Long, Integer> allChatRoomSessions = webSocketSessionService.countChatRoomSessions(joiningUsername);
 
         if(allChatRoomSessions.containsKey(chatId) && allChatRoomSessions.get(chatId) == 1) {
             log.trace("Sending join message to chat room: {}", chatId);
-            chatService.updateLastMessageSeen(chatId, joiningUsername);
+            chatService.markMessagesAsSeen(chatId, joiningUsername);
             sendMessages(chatRoom, new ChatNotificationJoined(chatId, joiningUsername));
         }
     }
 
-    public void onUserDisconnect(String leavingUsername) {
-        Set<String> subIds = new HashSet<>(chatSessionTracker.getSubscriptionIds());
-        Map<Long, Integer> allChatRoomSessions = countChatRoomSessions(leavingUsername);
+    public void onUserDisconnect(String leavingUsername, Map<String, Long> subscriptions) {
```

Comment:
to jest jedyne miejsce wykorzystania tego chatSessionTracker. Z tego co rozumiem to przy disconnect potrzebujemy miec infromacje o chatId, do ktorych byl podlaczony. Ale na upartego mozna uzyc czegos w stylu metody z webSocketSessionService: countChatRoomSessions i w tej metodzie przejsc po wszystkich subskrubcjach uzytkownika i zebrac wszystkie chatId, do ktorych byl podlaczony w bardzo podobny sposob jak w obecnej metodzie. W sumie to nawet sama metoda zwraca te chatId wraz z liczba sesji na czacie. Wiec nwm czy metoda nie bylaby wygladac np. tak:
```suggestion
        public void onUserDisconnect(String leavingUsername) {
        Map<Long, Integer> allChatRoomSessions = webSocketSessionService.countChatRoomSessions(leavingUsername);
        
        for(Long chatId: allChatRoomSessions.keySet()) {
            onUserUnsubscribe(leavingUsername, chatId, allChatRoomSessions);
        }

    }
```

Reply:
albo zostawic tego trackera, bo tak naprawde potrzebujemy informacji o czatach i liczbach sesji dla czatow, to moze zmienic mape z 
private final Map<String, Long> subsToChatId na private final Map<Long, List<String>> subsToChatId i wtedy dla kazdego czatu przechowywac liste otwartych sesji i nie potrzeba juz metody: webSocketSessionService.countChatRoomSessions, bo liczbe mozna zrobic biorac dlugosc listy

Reply:
nawet mozna zostawic w tracker taka mape jaka jest i na jej podstawie liczyc allChatRoomSessions skoro mamy juz tego trackera. Albo tak jak mowie usunac go. Moze czegos nie widze, wiec czekam co odpowiesz xd

Reply:
> to jest jedyne miejsce wykorzystania tego chatSessionTracker. Z tego co rozumiem to przy disconnect potrzebujemy miec infromacje o chatId, do ktorych byl podlaczony. Ale na upartego mozna uzyc czegos w stylu metody z webSocketSessionService: countChatRoomSessions i w tej metodzie przejsc po wszystkich subskrubcjach uzytkownika i zebrac wszystkie chatId, do ktorych byl podlaczony w bardzo podobny sposob jak w obecnej metodzie. W sumie to nawet sama metoda zwraca te chatId wraz z liczba sesji na czacie. Wiec nwm czy metoda nie bylaby wygladac np. tak:

Właśnie pierwotnie coś takiego było, ale okazuje się, że na evencie disconnect spring już nie ma informacji o subskrypcji więc `allChatRoomSessions` będzie puste

>  private final Map<String, Long> subsToChatId na private final Map<Long, List> subsToChatId i wtedy dla kazdego czatu przechowywac liste otwartych sesji i nie potrzeba juz metody: webSocketSessionService.countChatRoomSessions, bo liczbe mozna zrobic biorac dlugosc listy

Można. Może nawet szybsze będzie

Reply:
aa ok, to git i co do drugiego tez git

Reply:
A jednak jest problem. Bo zdałem sobie sprawę, że ten tracker to ma subskrypcje użytkownika w obecnej sesji przez scope na websocket. Więc i tak będę musiał liczyć te `allChatRoomSessions` bo potrzebuję wszyskie subskrypcje ze wszystkich sesji użytkownika.

W teorii mógłbym zrobić osobną klasę co trzyma to centralnie, ale wrócilibyśmy do poprzedniego rozwiązania a obecne jest o tyle gorsze, że trzeba id wyciągać z destination ale imo to nie jest aż tak wielki narzut

Reply:
No tak po jak usuniesz scope websocket to jeszcze tą mape musisz rozrozniac miedzy uzytkownikami.
A to nie lepiej wgl wywalic tego trackera i na podstawie tej metody `allChatRoomSessions ` zbierac co nam potrzebne?

Reply:
a dobra bo spring nie ma informacji